### ISSUE TITLE
Very poor results from GPU implementation using CNN Tutorial

### ISSUE THREAD

**Issue by mattlyon93** [tensorflow/tensorflow#45212]
**System information**
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No, I have used the stock code provided by [The CNN Tutorial](https://www.tensorflow.org/tutorials/images/cnn).
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04):** Windows 10 Pro, Version 20H2
- **TensorFlow installed from (source or binary):** TensorFlow installed via pip command `pip install tensorflow`
- **TensorFlow version (use command below):** v2.3.0-54-gfcc4b966f1 2.3.1
- **Python version:** 3.8.6
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- **CUDA/cuDNN version:** CUDA v10.1, cuDNN v7.6.0.64
- **GPU model and memory:** NVIDIA GeForce RTX 3080, 10GB

**Describe the current behaviour**
I followed the CNN tutorial linked above, using both GPU and CPU implementations. Whilst the CPU implementation produced very similar results to the graphs displayed on the tutorial, the GPU implementation never surpassed 0.1 accuracy in training.

I enabled the CPU implementation by the following line of code:
```python
import os
os.environ['CUDA_VISIBLE_DEVICES'] = '-1'
```

I discovered this bug by testing and training my own implementation which, similarly, had good results using CPU and terrible results using GPU.

**Describe the expected behaviour**
I expect the GPU implementation results to closely match that of the CPU implementation results.

**Standalone code to reproduce the issue**
Link to the jupyter notebook can be found [here](https://drive.google.com/file/d/1C6bnXhBqHCv-rNqS1QWWf4yZx-6x-qte/view?usp=sharing).

With minimal code below:
```python
import os
import tensorflow as tf
from tensorflow.keras import datasets, layers, models

(train_images, train_labels), (test_images, test_labels) = datasets.cifar10.load_data()

train_images, test_images = train_images / 255.0, test_images / 255.0

model = models.Sequential()
model.add(layers.Conv2D(32, (3, 3), activation='relu', input_shape=(32, 32, 3)))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(64, (3, 3), activation='relu'))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(64, (3, 3), activation='relu'))
model.add(layers.Flatten())
model.add(layers.Dense(64, activation='relu'))
model.add(layers.Dense(10))

model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])

history = model.fit(train_images, train_labels, epochs=10, 
                    validation_data=(test_images, test_labels))
```

**Other info / logs**
Below is the output from the `model.fit` step in the code provided above.

> Epoch 1/10
1563/1563 [==============================] - 4s 3ms/step - loss: 2.3028 - accuracy: 0.0960 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 2/10
1563/1563 [==============================] - 4s 2ms/step - loss: 2.3028 - accuracy: 0.0975 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 3/10
1563/1563 [==============================] - 4s 2ms/step - loss: 2.3028 - accuracy: 0.0968 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 4/10
1563/1563 [==============================] - 4s 2ms/step - loss: 2.3028 - accuracy: 0.0967 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 5/10
1563/1563 [==============================] - 4s 3ms/step - loss: 2.3027 - accuracy: 0.0990 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 6/10
1563/1563 [==============================] - 4s 3ms/step - loss: 2.3028 - accuracy: 0.0970 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 7/10
1563/1563 [==============================] - 4s 3ms/step - loss: 2.3028 - accuracy: 0.0972 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 8/10
1563/1563 [==============================] - 4s 2ms/step - loss: 2.3028 - accuracy: 0.0980 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 9/10
1563/1563 [==============================] - 4s 2ms/step - loss: 2.3028 - accuracy: 0.1002 - val_loss: 2.3026 - val_accuracy: 0.1000
Epoch 10/10
1563/1563 [==============================] - 4s 2ms/step - loss: 2.3028 - accuracy: 0.1007 - val_loss: 2.3026 - val_accuracy: 0.1000

The python process that was running the jupyter notebook displayed this warning:
> 2020-11-26 17:44:27.891143: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cublas64_10.dll
library cudnn64_7.dll
ation is supported on Cuda non-Windows platforms only
Relying on driver to perform ptx compilation.
Modify $PATH to customize ptxas location.
This message will be only logged once.

**Comment by amahendrakar** [comment]
@mattlyon93,
I did not observe much difference in the accuracy while running the code on [CPU](https://colab.research.google.com/gist/amahendrakar/eadf44fb38c5d44b715cfe7ea28ed5f1/45212-cpu.ipynb) and [GPU](https://colab.research.google.com/gist/amahendrakar/a77a13624c25bed71b3bd47c5d4286db/45212-gpu.ipynb). Please check the attached gist for reference. Thanks!

**Comment by mattlyon93** [comment]
It turns out the issue was due to the version of CUDA. Reading through comments in issue [#43588](https://github.com/tensorflow/tensorflow/issues/43588), it seems that the RTX 3080 is only compatible with CUDA 11.1+, therefore I upgraded to the latest cuDNN and CUDA versions, and installing `tf-nightly`.

This fixed the problem.

**Comment by mattlyon93** [comment]
I have fixed this issue by installing the following: `CUDA v11.0`, `cuDNN 8.0.2`, `tensorflow v2.4.0rc3`.
